var/spack/repos/builtin/packages/cosma/package.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Cosma(CMakePackage, CudaPackage):
+    """
+    Distributed Communication-Optimal Matrix-Matrix Multiplication Library
+    """
+
+    homepage = "https://github.com/eth-cscs/COSMA"
+    maintainers = ['teonnik', 'kabicm']
+
+    # note: The default archives produced with github do not have the archives
+    #       of the submodules.
+    version('develop',
+            git='https://github.com/eth-cscs/COSMA.git',
+            branch='master',
+            submodules=True)
+    version('2.0.2',
+            url='https://github.com/eth-cscs/COSMA/releases/download/v2.0.2/cosma.tar.gz',
+            sha256='4f3354828bc718f3eef2f0098c3bdca3499297497a220da32db1acd57920c68d')
+    # note: this version fails to build at the moment
+    # version('1.0.0',
+    #         url='https://github.com/eth-cscs/COSMA/releases/download/1.0/cosma.tar.gz',
+    #         sha256='c142104258dcca4c17fa7faffc2990a08d2777235c7980006e93c5dca51061f6')
+
+    variant('blas', default='mkl',
+            values=('mkl', 'openblas', 'netlib', 'cuda', 'rocm'),
+            description='BLAS backend')
+    variant('scalapack', default='none',
+            values=('mkl', 'netlib'),
+            description='Optional ScaLAPACK support.')
```

Comment:
The way Spack handles these is through virtual dependencies. Basically, you add a dependency on `blas` or `scalapack`, and then you set your default BLAS/LAPACK/SCALAPACK provider in `packages.yaml` or on the command-line. So you can remove these variants.